src/services/UserValidationService.ts

```diff
@@ -0,0 +1,60 @@
+import { Service } from '@tsed/di';
+import { Unauthorized } from '@tsed/exceptions';
+import * as bcrypt from 'bcrypt';
+
+import { User } from '../entities/User';
+
+@Service()
+export class UserValidationService {
+
+	public async validateUser(
+		userToValidate: User, 
+	): Promise<User> {
+
+		const username: string = userToValidate.username;
+		const email: string= userToValidate.email;
+		const password: string = userToValidate.password;
+		
+		if (Object.keys(userToValidate).length === 0) {
+			throw new Unauthorized("Empty request body!");
+		}
+	
+		if (!username || !email || !password) {
+			throw new Unauthorized("Missing credentials!");
+		}
+
+		if (!this.containWhitespaces(username)) {
+			throw new Unauthorized("Whitespaces in username!");
+		}
+
+		if (!this.isValidEmailFormat(email)) {
+			throw new Unauthorized("Invalid email format!");
+		}
+
+		if (!this.isStrongPassword(password)) {
+			throw new Unauthorized("Week password!");
+		}
+
+		userToValidate.password = await this.encryptPassword(password);
```

Comment:
Inside function `validateUser`, you are encrypting password. This is likely misleading and encrypting should be called outside of validation - encrypting password is not a validation.